src/renderer/widgets/DelegationModal/components/AddCustomDelegationModal.tsx

```diff
@@ -0,0 +1,65 @@
+import { useUnit } from 'effector-react';
+
+import { useI18n } from '@/app/providers';
+import { useModalClose } from '@/shared/lib/hooks';
+import { Step, isStep, validateAddress } from '@/shared/lib/utils';
+import { BaseModal, Button, Icon, Identicon, Input } from '@/shared/ui';
+import { OperationTitle } from '@/entities/chain';
+import { networkSelectorModel } from '@/features/governance';
+import { delegationModel } from '../model/delegation-model';
+
+export const AddCustomDelegationModel = () => {
+  const { t } = useI18n();
+
+  const step = useUnit(delegationModel.$step);
+  const chain = useUnit(networkSelectorModel.$governanceChain);
+  const customDelegate = useUnit(delegationModel.$customDelegate);
+  const isValidAddress = validateAddress(customDelegate || '');
```

Comment:
since we use customDelegate as  `customDelegate ||  ''` everywhere maybe put it in the model as 
`const $customDelegate = restore(customDelegateChanged, '').reset(openCustomModal);`